compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -2573,11 +2582,11 @@ impl<'a> Parser<'a> {
     }
 
     pub(super) fn parse_arm(&mut self) -> PResult<'a, Arm> {
-        fn check_let_expr(expr: &Expr) -> (bool, bool) {
+        fn check_let(expr: &Expr) -> (bool, bool) {
             match expr.kind {
                 ExprKind::Binary(_, ref lhs, ref rhs) => {
```

Comment:
nit: should this be considering if the binary op is  `&&`? Not exactly sure how this code works and what it's protecting against -- could you leave a comment?